src/occur.ml

```diff
@@ -0,0 +1,42 @@
+(** Some tools to encode the position of subterms in a term. *)
+open Extra
+
+(** Each element of the list is a path in the tree of the term.  For instance,
+    in the term [Appl(Appl(f, x), Appl(Appl(g, a), b))], the subterm [a] has
```

Comment:
Does it?

Reply:
Well, `a` appears quite deep in the term (three levels), yet, the path contains only two numbers.

Reply:
For me `a` should be at position "1.0.1"; not `1.0`.

Reply:
And `x` should be at position "0.1" I guess.

Reply:
I see the following tree
```
         x
      / 
f - +              a
      \          /
         g - +
                 \
                   b
```
`f` at `eps`, `x` at `0`, `g` at `1`, `a` at `1.0` and `b` at `1.1`

Reply:
OK, I see: you are not working with binary application nodes. I saw the following tree (which corresponds more closely to the term representation, and is also more usual in some way).
```
     @
    / \
   @   \
  / \   \
 f   x   @
        / \
       @   b
      / \
     g   a
```

Please, add a word explaining that the number `i` means "going under the `i-1`-th argument. It might be worth changing the example so that `f` has more than two arguments to make this clearer.

Reply:
Or simply remove the `Appl`'s and write [f x (g a b)].

Reply:
The representation I used is more usual in rewriting I think

Reply:
No.

Reply:
[f x (g a b)] is the usual first-order (uncurried) representation of terms. Using an App node is the usual higher-order (curried) representation of terms (in lambda-calculus). Internally, we use an App node. But your notion of position is the first-order one.